ecs/pod_creator.go

```diff
@@ -27,11 +31,235 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
+		}
+		return nil, errors.Wrap(catcher.Resolve(), "running task")
+	}
+
+	if len(runOut.Tasks) == 0 || runOut.Tasks[0].TaskArn == nil {
+		return nil, errors.New("expected a task to be running in ECS, but none was returned")
+	}
+
+	resources := cocoa.NewECSPodResources().
+		SetCluster(*mergedPodExecutionOpts.Cluster).
+		SetSecrets(translatePodSecrets(secrets)).
+		SetTaskDefinition(*taskDef).
+		SetTaskID(*runOut.Tasks[0].TaskArn)
+
+	options := NewBasicECSPodOptions().
+		SetClient(m.client).
+		SetVault(m.vault).
+		SetStatus(cocoa.Running).
+		SetResources(*resources)
+
+	p, err := NewBasicECSPod(options)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "creating pod")
+	}
+
+	return p, nil
 }
 
 // CreatePodFromExistingDefinition creates a new pod backed by AWS ECS from an
 // existing definition.
 func (m *BasicECSPodCreator) CreatePodFromExistingDefinition(ctx context.Context, def cocoa.ECSTaskDefinition, opts ...*cocoa.ECSPodExecutionOptions) (cocoa.ECSPod, error) {
 	return nil, errors.New("TODO: implement")
 }
+
+// exportTags converts strings into ECS tags.
+func exportTags(tags []string) []*ecs.Tag {
+	ecsTags := []*ecs.Tag{}
+
+	for _, tag := range tags {
+		ecsTag := &ecs.Tag{}
+		ecsTag.SetKey(tag)
+		ecsTags = append(ecsTags, ecsTag)
+	}
+
+	return ecsTags
+}
+
+// exportStrategy converts the strategy and parameter into an ECS placement strategy.
+func exportStrategy(strategy *cocoa.ECSPlacementStrategy, param *cocoa.ECSStrategyParameter) []*ecs.PlacementStrategy {
+	placementStrat := ecs.PlacementStrategy{}
+
+	placementStrat.SetField(utility.FromStringPtr(param))
+
+	if strategy == nil {
+		placementStrat.SetType("")
+	} else {
+		placementStrat.SetType(string(*strategy))
+	}
+
+	return []*ecs.PlacementStrategy{&placementStrat}
+}
+
+// exportEnvVars converts the non-secret environment variables into ECS environment variables.
+func (m *BasicECSPodCreator) exportEnvVars(variables []cocoa.EnvironmentVariable) []*ecs.KeyValuePair {
+	keyValuePairs := []*ecs.KeyValuePair{}
+
+	for _, variable := range variables {
+		if variable.SecretOpts == nil {
+			keyValue := ecs.KeyValuePair{}
+			keyValue.SetName(utility.FromStringPtr(variable.Name)).SetValue(utility.FromStringPtr(variable.Value))
+			keyValuePairs = append(keyValuePairs, &keyValue)
+		}
+	}
+
+	return keyValuePairs
+}
+
+// getSecrets retrieves the secrets from the secret environment variables for the pod.
+func (m *BasicECSPodCreator) getSecrets(merged cocoa.ECSPodCreationOptions) []cocoa.SecretOptions {
+	secrets := []cocoa.SecretOptions{}
+
+	for _, def := range merged.ContainerDefinitions {
+		for _, variable := range def.EnvVars {
+			if variable.SecretOpts != nil {
+				secrets = append(secrets, *variable.SecretOpts)
+			}
+		}
+	}
+
+	return secrets
+}
+
+// createSecrets creates secrets that do not already exist.
+func (m *BasicECSPodCreator) createSecrets(ctx context.Context, secrets []cocoa.SecretOptions) error {
+	if m.vault == nil {
```

Comment:
This should probably be moved to L181. We don't need to have a vault unless we need to create secrets.